src/handle_action.cpp

```diff
@@ -753,24 +755,92 @@ static void smash()
 
-            const int mod_sta = 2 * u.get_standard_stamina_cost();
+            // Do damage to hands/feet if target is hard and we used hands/feet
+            const bool hard_target = ( ( bash_furn > 2 ) || ( bash_furn == -1 && bash_ter > 2 ) );
+
+            if( hard_target ) {
+                int glove_coverage = 0;
+                int shoe_coverage = 0;
+
+                for( const item &i : u.worn ) {
+                    int temp_coverage = i.get_coverage();
+
+                    // Feet coverage
+                    if( ( i.covers( bodypart_id( "foot_l" ) ) || i.covers( bodypart_id( "foot_r" ) ) ) ) {
+                        if( shoe_coverage < temp_coverage ) {
+                            shoe_coverage = temp_coverage;
+                        }
+                    }
+                    // Hand coverage
+                    if( ( i.covers( bodypart_id( "hand_l" ) ) || i.covers( bodypart_id( "hand_r" ) ) ) ) {
+                        if( glove_coverage < temp_coverage ) {
+                            glove_coverage = temp_coverage;
+                        }
+                    }
+                }
+
+                // If the player has no weapon or has a weapon with 0 bash.
+                // the second case acts like an automatic force unarmed.
+                if( !w->damage_melee( DT_BASH ) ) {
+                    w_bash = false;
+
+                    // Determine hand/foot damage. 90% coverage is the maximum damage reduction.
+                    const int dam_hand = roll_remainder( 5.0 * std::max( 0.0, 1 - glove_coverage / 90.0 ) );
+                    const int dam_foot = roll_remainder( 5.0 * std::max( 0.0, 1 - shoe_coverage / 90.0 ) );
+
+                    // Determine hand/foot with highest hp
+                    bodypart_id  bp_hand_id = ( u.hp_cur[hp_arm_r] > u.hp_cur[hp_arm_l] ) ? bodypart_id( "hand_r" ) :
+                                              bodypart_id( "hand_l" );
+                    bodypart_id bp_foot_id = ( u.hp_cur[hp_leg_r] > u.hp_cur[hp_leg_l] ) ? bodypart_id( "foot_r" ) :
+                                             bodypart_id( "foot_l" );
+
+                    // Pre-determine limb hp after damage is applied.
+                    const int foot_hp_calc = u.hp_cur[u.bp_to_hp( bp_foot_id.obj().token )] - dam_foot;
+                    const int hand_hp_calc = u.hp_cur[u.bp_to_hp( bp_hand_id.obj().token )] - dam_hand;
+
+                    // If we have a 2 handed weapon with 0 bash, use feet.
+                    // Otherwise, use the highest resultant limb hp to smash with.
+                    bool use_hands = w->is_two_handed( u ) ? false : hand_hp_calc > foot_hp_calc;
+
+                    // If the player has a weapon, do not smash with our hands/feet
+                    // if our best option will break our limb. Use weapon instead.
+                    if( !w->is_null() ) {
+                        w_bash = use_hands ? ( !hand_hp_calc > 0 ) : ( !foot_hp_calc > 0 );
```

Comment:
Why so complicated? `!hand_hp_calc > 0` == `( hand_hp_calc == 0 ) > 0` == `( hand_hp_calc == 0 ) == true` == `hand_hp_calc == 0`
```suggestion
                        w_bash = use_hands ? hand_hp_calc == 0 : foot_hp_calc == 0;
```
Or even just:
```suggestion
                        w_bash = ( use_hands ? hand_hp_calc : foot_hp_calc ) == 0;
```

Reply:
In that thought simplification, it would lose the handling of negative numbers. hand_hp_calc and foot_hp_calc doesn't have 0 as the lower bound. If this were used, the player would be smashing with their already broken hands or feet.

Reply:
You current code is handling negative numbers like any positive number.

`( !hand_hp_calc > 0 )` is equivalent to `( ( hand_hp_calc != 0 ) > 0 )`.

Reply:
That's fair, it should just be ```hand_hp_calc > 0``` then. Updated to just that